src/foundry/common/prosemirror/highlight-matches-plugin.d.mts

```diff
@@ -0,0 +1,82 @@
+import { Schema } from "prosemirror-model";
+import type ProseMirrorPlugin from "./plugin.d.mts";
+import type { ProseMirrorMenu } from "./menu.d.mts";
+import { EditorState, Plugin } from "prosemirror-state";
+import { EditorView } from "prosemirror-view";
+import type { EmptyObject } from "../../../types/utils.d.mts";
+
+declare namespace ProseMirrorHighlightMatchesPlugin {
+  /**
+   * A class responsible for handling the display of automated link recommendations when a user highlights text in a
+   * ProseMirror editor.
+   */
+  class PossibleMatchesTooltip {
+    /**
+     * @param view - The editor view
+     */
+    constructor(view: EditorView);
+
+    /**
+     * A reference to any existing tooltip that has been generated as part of a highlight match.
+     */
+    tooltip: HTMLElement;
+
+    update(view: EditorView, lastState: EditorState): Promise<void>;
+
+    /**
+     * Create a locked tooltip at the given position
+     * @param position - A position object with coordinates for where the tooltip should be placed
+     * @param text     - Explicit tooltip text or HTML to display.
+     * @param options  - Additional options which can override tooltip behavior.
+     */
+    _createTooltip(
+      position: {
+        /** Explicit top position for the tooltip */
+        top?: string;
+        /** Explicit right position for the tooltip */
+        right?: string;
+        /** Explicit bottom position for the tooltip */
+        bottom?: string;
+        /** Explicit left position for the tooltip */
+        left?: string;
```

Comment:
These are unfortunately properties despite being within a function. Probably wrap the object in in `InexactPartial`.